tutorials/tools/binary-to-decimal.md

```diff
@@ -0,0 +1,146 @@
+A [binary number](https://en.wikipedia.org/wiki/Binary_number) is a number expressed in base-2, as opposed to conventional decimal numbers in base-10.
+
+Below is a live calculator.
+
+<style>
+  #binary {
+    font-size: 1.5em;
+    padding: 0.25em;
+    border-radius: 4px;
+  }
+
+  #result {
+    display: inline-block;
+    font-size: 1.5em;
+    padding: 0.25em;
+    border-radius: 4px;
+    background-color: #ddd;
+    width: 310px;
+    overflow-wrap: anywhere;
+  }
+
+  #right-arrow {
+    font-size: 2em;
+    padding: 0.25em;
+    padding-left: 0.5em;
+    padding-right: 0.5em;
+  }
+
+  @media (max-width: 1000px) {
+    #decimal {
+      width: 100%;
+    }
+
+    #result {
+      width: 8em;
+    }
+  }
+</style>
+
+<table>
+  <thead>
+    <th>Binary</th>
+    <th></th>
+    <th>Decimal</th>
+  </thead>
+  <tr>
+    <td>
+      <input type="number" id="binary" placeholder="101010" />
+    </td>
+    <td id="right-arrow">
+      &#10140;
+    </td>
+    <td>
+      <div id="result"></div>
+    </td>
+  </tr>
+</table>
+<script>
+  document.querySelector('#result').innerHTML = parseInt(101010, 2);
+  document.querySelector('#binary').addEventListener('change', function() {
+    const value = parseInt(document.querySelector('#binary').value, 2);
+    if (value == null || isNaN(value)) {
+      return;
+    } 
+    document.querySelector('#result').innerHTML = value;
+  });
+  document.querySelector('#binary').addEventListener('keyup', function() {
+    const value = parseInt(document.querySelector('#binary').value, 2);
+    if (value == null || isNaN(value)) {
+      return;
+    } 
+    document.querySelector('#result').innerHTML = value;
+  });
+  document.querySelector('#binary').focus();
+</script>
+
+## How the Calculator Works
+
+Converting binary numbers to decimal is easy.
+For example, `let x = 101010` creates a new variable `x` that contains the number `101010`.
```

Comment:
`let x = '10101'`. Techically `let x = 10101` also works, but that means `x` is a number equal to 10101 in base 10, not base 2. Use strings, not numbers